app/Console/Commands/DeviceAdd.php

```diff
@@ -41,19 +35,27 @@ public function __construct()
         parent::__construct();
 
         $this->optionValues = [
-            'transport' => ['udp', 'udp6', 'tcp', 'tcp6'],
-            'port-association-mode' => [PortAssociationMode::class, 'getModes'],
-            'auth-protocol' => [\LibreNMS\SNMPCapabilities::class, 'supportedAuthAlgorithms'],
-            'privacy-protocol' => [\LibreNMS\SNMPCapabilities::class, 'supportedCryptoAlgorithms'],
+            'transport' => [['udp', 'udp6', 'tcp', 'tcp6']],
+
+            'port-association-mode' => [[PortAssociationMode::class, 'getModes']],
+            'auth-protocol' => [[\LibreNMS\SNMPCapabilities::class, 'supportedAuthAlgorithms']],
+            'privacy-protocol' => [[\LibreNMS\SNMPCapabilities::class, 'supportedCryptoAlgorithms']],
+        ];
+
+        $this->optionDefaults = [
+            'port' => [[Config::class, 'get'], ['snmp.port', 161]],
```

Comment:
You made this way more complex.  Just use an anonymous function:
```php
'port' => fn () => Config::get('snmp.port', 161),
```

Except we can't use that style yet :D

Reply:
Slightly better now :)